components/nexusoperations/executors.go

```diff
@@ -298,6 +327,19 @@ func (e taskExecutor) saveResult(ctx context.Context, env hsm.Environment, ref h
 				return hsm.TransitionOutput{}, err
 			}
 			if result.Pending != nil {
+				var links []*commonpb.Link
+				for _, nexusLink := range result.Links {
+					switch nexusLink.Type {
+					case string((&commonpb.Link{}).ProtoReflect().Descriptor().FullName()):
+						link := &commonpb.Link{}
+						if err := proto.Unmarshal(nexusLink.Data, link); err != nil {
+							return hsm.TransitionOutput{}, err
+						}
+						links = append(links, link)
+					default:
+						return hsm.TransitionOutput{}, fmt.Errorf("invalid link data type: %q", nexusLink.Type)
```

Comment:
I'd probably just log an error and skip this link in the response just to be safe. Alternatively you can ensure that this translates into a non retryable error but I would rather be more forgiving than break the operation.